go/vt/vttest/environment.go

```diff
@@ -17,34 +18,262 @@ limitations under the License.
+		flavor = os.Getenv("MYSQL_FLAVOR")
 	}
-	return path.Join(vttop, "py/vttest/run_local_database.py"), nil
+
+	if flavor == "" {
+		flavor = DefaultMySQLFlavor
+	}
+
+	mycnf := []string{"config/mycnf/vtcombo.cnf"}
+	switch flavor {
+	case "MariaDB":
+		mycnf = append(mycnf, "config/mycnf/default-fast.cnf")
+		mycnf = append(mycnf, "config/mycnf/master_mariadb.cnf")
+
+	case "MySQL56":
+		mycnf = append(mycnf, "config/mycnf/default-fast.cnf")
+		mycnf = append(mycnf, "config/mycnf/master_mysql56.cnf")
+
+	default:
+		return "", nil, fmt.Errorf("unknown mysql flavor: %s", flavor)
+	}
+
+	for i, cnf := range mycnf {
+		mycnf[i] = path.Join(os.Getenv("VTTOP"), cnf)
+	}
+
+	return flavor, mycnf, nil
+}
+
+// EnvVars implements EnvVars for LocalTestEnv
+func (env *LocalTestEnv) EnvVars() []string {
+	return env.Env
+}
+
+// BinaryPath implements BinaryPath for LocalTestEnv
+func (env *LocalTestEnv) BinaryPath(binary string) string {
+	return path.Join(os.Getenv("VTROOT"), "bin", binary)
+}
+
+// MySQLManager implements MySQLManager for LocalTestEnv
+func (env *LocalTestEnv) MySQLManager(mycnf []string, snapshot string) (MySQLManager, error) {
+	return &Mysqlctl{
+		Binary:    env.BinaryPath("mysqlctl"),
+		InitFile:  path.Join(os.Getenv("VTTOP"), "config/init_db.sql"),
+		Directory: env.TmpPath,
+		Port:      env.PortForProtocol("mysql", ""),
+		MyCnf:     append(mycnf, env.DefaultMyCnf...),
+		Env:       env.EnvVars(),
+	}, nil
 }
 
-func vtgateProtocol() string {
+// DefaultProtocol implements DefaultProtocol for LocalTestEnv.
+// It is always GRPC.
+func (env *LocalTestEnv) DefaultProtocol() string {
 	return "grpc"
 }
 
-// randomPort returns a random number between 10k & 30k.
+// PortForProtocol implements PortForProtocol for LocalTestEnv.
+func (env *LocalTestEnv) PortForProtocol(name, proto string) int {
+	switch name {
+	case "vtcombo":
+		if proto == "grpc" {
+			return env.BasePort + 1
+		}
+		return env.BasePort
+
+	case "mysql":
+		return env.BasePort + 2
+
+	case "vtcombo_mysql_port":
+		return env.BasePort + 3
+
+	default:
+		panic("unknown service name: " + name)
+	}
+}
+
+// ProcessHealthCheck implements ProcessHealthCheck for LocalTestEnv.
+// By default, it performs no service-specific health checks
+func (env *LocalTestEnv) ProcessHealthCheck(name string) HealthChecker {
+	return nil
+}
+
+// VtcomboArguments implements VtcomboArguments for LocalTestEnv.
+func (env *LocalTestEnv) VtcomboArguments() []string {
+	return []string{
+		"-service_map", strings.Join(
+			[]string{"grpc-vtgateservice", "grpc-vtctl"}, ",",
+		)}
+}
+
+// LogDirectory implements LogDirectory for LocalTestEnv.
+func (env *LocalTestEnv) LogDirectory() string {
+	return path.Join(env.TmpPath, "logs")
+}
+
+// Directory implements Directory for LocalTestEnv.
+func (env *LocalTestEnv) Directory() string {
+	return env.TmpPath
+}
+
+// TearDown implements TearDown for LocalTestEnv
+func (env *LocalTestEnv) TearDown() error {
+	return os.RemoveAll(env.TmpPath)
```

Comment:
@alainjobart says that it's better to SIGTERM vtcombo and wait for it to exit. This way, all logs will be flushed, etc.